Author: idra
Revision: r23356
Modified: source/lib/util_str.c /branches/SAMBA_3_0_25/source/lib/util_str.c /branches/SAMBA_3_0_26/source/lib/util_str.c
Added: 
Removed: 



We missed to add the &#39;c&#39; character to the list of valid ones for
shell escaping.

I hate this kind of bugs more than how Jeremy hates off by ones :(

Simo.


Index: source/lib/util_str.c
===================================================================
--- source/lib/util_str.c	(revision 23355)
+++ source/lib/util_str.c	(revision 23356)
@@ -2622,7 +2622,7 @@
  of characters. UNIX charset format.
 *******************************************************************/
 
-#define INCLUDE_LIST "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabdefghijklmnopqrstuvwxyz_/ \t.,"
+#define INCLUDE_LIST "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_/ \t.,"
 #define INSIDE_DQUOTE_LIST "$`\n\"\\"
 
 char *escape_shell_string(const char *src)

-------------------------------------------------------------------
Author: jra
Revision: r23346
Modified: source/smbd/nttrans.c /branches/SAMBA_3_0_25/source/smbd/nttrans.c /branches/SAMBA_3_0_26/source/smbd/nttrans.c
Added: 
Removed: 


Fix offline caching with XP/Vista. It was an off-by-one
in storing the access mask. I shouldn&#39;t have made this
mistake. Damn. Fixes bug #4673.
Jeremy


Index: source/smbd/nttrans.c
===================================================================
--- source/smbd/nttrans.c	(revision 23345)
+++ source/smbd/nttrans.c	(revision 23346)
@@ -420,7 +420,7 @@
 	p += 4;
 
 	if (flags & EXTENDED_RESPONSE_REQUIRED) {
-		p += 26;
+		p += 25;
 		SIVAL(p,0,FILE_GENERIC_ALL);
 		/* 
 		 * For pipes W2K3 seems to return
@@ -950,7 +950,7 @@
 
 	if (flags & EXTENDED_RESPONSE_REQUIRED) {
 		uint32 perms = 0;
-		p += 26;
+		p += 25;
 		if (fsp->is_directory || can_write_to_file(conn, fname, &sbuf)) {
 			perms = FILE_GENERIC_ALL;
 		} else {
@@ -1035,7 +1035,7 @@
 	p += 4;
 	
 	if (flags & EXTENDED_RESPONSE_REQUIRED) {
-		p += 26;
+		p += 25;
 		SIVAL(p,0,FILE_GENERIC_ALL);
 		/* 
 		 * For pipes W2K3 seems to return
@@ -1631,7 +1631,7 @@
 
 	if (flags & EXTENDED_RESPONSE_REQUIRED) {
 		uint32 perms = 0;
-		p += 26;
+		p += 25;
 		if (fsp->is_directory || can_write_to_file(conn, fname, &sbuf)) {
 			perms = FILE_GENERIC_ALL;
 		} else {

